dateparser/search_dates/search.py

```diff
@@ -0,0 +1,278 @@
+                    text=date_object_candidate,
+                    parser=parser,
+                    datetime_object=parsed_date_object.date_obj,
+                )
+
+            returnable_objects.append(
+                (date_object_candidate.strip(" .,:()[]-'"), parsed_date_object.date_obj)
+            )
+
+            if deep_search:
+                start_index = text.find(date_object_candidate)
+                end_index = start_index + len(date_object_candidate)
+                if start_index < 0:
+                    reduced_text_candidate = None
+                else:
+                    reduced_text_candidate = text[:start_index] + text[end_index:]
+            break
+        else:
+            for splitter in _secondary_splitters:
+                secondary_split = re.split(
+                    "(?<! )[" + splitter + "]+(?! )", date_object_candidate
+                )
+                if secondary_split and len(secondary_split) > 1:
+                    reduced_text_candidate = " ".join(secondary_split)
+                    secondary_split_made = True
+
+            if not reduced_text_candidate:
+                is_previous_punctuation = False
+                for index, char in enumerate(date_object_candidate):
+                    if char in _punctuations:
+                        if is_previous_punctuation:
+                            double_punctuation_split = [
+                                text[: index - 1],
+                                text[index - 1:],
+                            ]
+                            reduced_text_candidate = " ".join(double_punctuation_split)
+                            break
+                        is_previous_punctuation = True
+                    else:
+                        is_previous_punctuation = False
+
+    if reduced_text_candidate:
+        reduced_text_candidate = reduced_text_candidate.strip(" .,:()[]-'")
+
+    if (deep_search or secondary_split_made) and not (
+        text == reduced_text_candidate and is_recursion_call
+    ):
+        if reduced_text_candidate and len(reduced_text_candidate) > 2:
+            returnable_objects = _joint_parse(
+                text=reduced_text_candidate,
+                parser=parser,
+                data_carry=returnable_objects,
+                is_recursion_call=True,
+            )
+
+    return returnable_objects
+
+
+class DateSearch:
+    """
+    Class which handles language detection, translation and subsequent generic parsing of
+    string representing date and/or time.
+
+    :return: A date search instance
+    """
+
+    def __init__(self):
+        self.search_languages = SearchLanguages()
+
+    @apply_settings
+    def search_parse(
+        self,
+        text,
+        language_shortname,
+        settings,
+        limit_date_search_results=None,
+        make_joints_parse=True,
+        deep_search=True,
+        accurate_return_text=False,
+    ) -> List[tuple]:
```

Comment:
Could you remove the typing for Python 3.5 support? :)

Reply:
Done 👍🏻